packages/@react-aria/color/src/useColorWheel.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {ColorWheelAriaProps, ColorWheelAriaResult, ColorWheelState} from '@react-types/color';
+import {focusWithoutScrolling, mergeProps} from '@react-aria/utils';
+import React, {useCallback, useRef} from 'react';
+import {useMove} from '@react-aria/interactions';
+
+function degToRad(deg: number) {
+  return deg * Math.PI / 180;
+}
+
+function radToDeg(rad: number) {
+  return rad * 180 / Math.PI;
+}
+
+// 0deg = 3 o'clock. increses clockwise
+function angleToCartesian(angle: number, radius: number): {x: number, y: number} {
+  let rad = degToRad(360 - angle + 90);
+  let x = Math.sin(rad) * (radius);
+  let y = Math.cos(rad) * (radius);
+  return {x, y};
+}
+function cartesianToAngle(x: number, y: number, radius: number): number {
+  let deg = radToDeg(Math.atan2(y / radius, x / radius));
+  return (deg + 360) % 360;
+}
+
+export function useColorWheel(props: ColorWheelAriaProps, state: ColorWheelState): ColorWheelAriaResult {
+  let {inputRef, containerRef, isDisabled, step = 1, innerRadius, outerRadius} = props;
+
+  let thumbRadius = (innerRadius + outerRadius) / 2;
+
+  let focusInput = useCallback(() => {
+    if (inputRef.current) {
+      focusWithoutScrolling(inputRef.current);
+    }
+  }, [inputRef]);
+
+  let stateRef = useRef<ColorWheelState>(null);
+  stateRef.current = state;
+
+  let currentPosition = useRef<{x: number, y: number}>(null);
+  let moveHandler = {
+    onMoveStart() {
+      currentPosition.current = null;
+    },
+    onMove({deltaX, deltaY, pointerType}) {
+      if (currentPosition.current == null) {
+        currentPosition.current = angleToCartesian(stateRef.current.value.getChannelValue('hue'), thumbRadius);
+      }
+      currentPosition.current.x += deltaX;
+      currentPosition.current.y += deltaY;
+      if (pointerType === 'keyboard') {
+        if (deltaX > 0) {
+          stateRef.current.setHue(stateRef.current.value.getChannelValue('hue') + step);
+        } else if (deltaX < 0) {
+          stateRef.current.setHue(stateRef.current.value.getChannelValue('hue') - step);
+        }
```

Comment:
I followed https://spectrum.adobe.com/page/color-wheel/#Keyboard-interactions, but page up/down and could also make sense here

Reply:
> Curious if arrow up/arrow down should do anything? cc. @majornista

* ArrowUp and ArrowRight should increment, ArrowLeft and ArrowDown should decrement. 
* PageUp and PageDown on a Hue color wheel makes sense to snap to intervals of 6.

something like:
```js
static const HUE_TICK_INTERVAL = 6;
const dir = deltaX < 0 ? -1 * step : step;
stateRef.current.setHue((Math[deltaX > 0 ? 'floor' : 'ceil'](stateRef.current.value.getChannelValue('hue') / HUE_TICK_INTERVAL) + dir) * HUE_TICK_INTERVAL);
```

Reply:
So like a "normal" slider? Should the behaviour of left/right swap with RTL?

Reply:
I don't think so in the case of a circle but unsure. We could ask design or the i18n team...

Reply:
What should page up/down do if the step size is bigger than 6?

Reply:
@mischnic That makes sense, but it might be helpful to snap to interval values that are factors of 360, so that page up/page down always resolves to 0/360.

Reply:
I've changed the logic so that you always get back to 0 (when going in the increasing direction) and so that you actually get to the same step stops when going in the other direction (because the distance between the last step and zero could also be smaller that the step size).